lib/inspec/config.rb

```diff
@@ -0,0 +1,286 @@
+# Represents InSpec configuration.  Merges defaults, config file options,
+# and CLI arguments.
+
+require 'pp'
+
+module Inspec
+  class Config
+    # These are options that apply to any transport
+    GENERIC_CREDENTIALS = %w{
+      backend
+      sudo
+      sudo_password
+      sudo_command
+      sudo_options
+      shell
+      shell_options
+      shell_command
+    }.freeze
+
+    extend Forwardable
+
+    # Many parts of InSpec expect to treat the Config as a Hash
+    def_delegators :@final_options, :each, :delete, :[], :[]=, :key?
+    attr_reader :final_options
+
+    def initialize(cli_opts = {}, cfg_io = nil, command_name = nil)
+      @command_name = command_name || (ARGV.empty? ? nil : ARGV[0].to_sym)
+      @defaults = Defaults.for_command(@command_name)
+
+      @cli_opts = cli_opts.dup
+      cfg_io = resolve_cfg_io(@cli_opts, cfg_io)
+      @cfg_file_contents = read_cfg_file_io(cfg_io)
+
+      @merged_options = merge_options
+      @final_options = finalize_options
+    end
+
+    def diagnose
```

Comment:
`pp` is fine here, but I'm wonder if the work you did with UI would prevent the need for it.

Reply:
There is nothing in the Inspec::UI work that deals with pretty-printing deep structures.